dnf5/main.cpp

```diff
@@ -503,6 +503,14 @@ void RootCommand::set_argument_parser() {
     installroot->link_value(&config.get_installroot_option());
     global_options_group->register_argument(installroot);
 
+    auto withmounts = parser.add_new_named_arg("with-mounts");
+    withmounts->set_long_name("with-mounts");
+    withmounts->set_description(
+        _("mount pseudo filesystems in ABSOLUTE_PATH when using installroot"));
```

Comment:
There is no built in way to handle the relationship that `--with-mounts` requires `--installroot=...` but we do have an `libdnf5::cli::ArgumentParserMissingDependentArgumentError` which I think it would be good to throw when the argument is used without installroot.
We already have options that do something similar.